File: Day301/findUniqueBinaryString.java
class Solution {
    public String findDifferentBinaryString(String[] nums) {
        int l = nums[0].length();
        int last = (int)Math.pow(2,l);
        Set<String> set = new HashSet<>();
        for (String num : nums) {
            set.add(num);
        }
        for (int i=last-1; i>=0; i--) {
            String check = Integer.toBinaryString(i);
            while (check.length()<l) {
                check ="0" +check;
            }
            // System.out.println(check);
            if (!set.contains(check)) {
                return check;
            }
        }
        return Integer.toBinaryString(nums.length);
    }
}

File: Day301/findUniqueBinaryString.go
func findDifferentBinaryString(nums []string) string {
    l := len(nums[0])
    n := int( math.Pow(float64(2), float64(l)))
    set := make(map[string]bool)
    for _, v := range nums {
        set[v] = true
    }
    for i:=0; i<n; i++ {
        check := strconv.FormatInt(int64(i), 2)
        for len(check) < l {
            check = "0" + check
        }
        if _,flag := set[check]; !flag {
            return check
        }
    }
    return strconv.FormatInt(int64(len(nums)), 2)
}